script.js

```diff
@@ -1,4 +1,306 @@
+      (userGuess == "paper" && itemName == "stone") ||
+      (userGuess == "stone" && itemName == "scissors")
+    ) {
+      playCount = playCount + 1;
+      userWinCount = userWinCount + 1;
+      var winningRate = (userWinCount / playCount) * 100;
+      var outcomeMessage = `You win!`;
+    } // if both parties choose the same object, it's a draw
+    else if (itemName == userGuess) {
+      playCount = playCount + 1;
+      drawCount = drawCount + 1;
+      winningRate = (userWinCount / playCount) * 100;
+      outcomeMessage = `It's a draw!`;
+    } // if the above two statements are false then the user has lost
+    else {
+      playCount = playCount + 1;
+      computerWinCount = computerWinCount + 1;
+      winningRate = (userWinCount / playCount) * 100;
+      outcomeMessage = `You lose!`;
+    }
+  }
+
+  // SPS game logic (reverse)
+  if (currentGameMode == "SPS" && reverseMode) {
+    if (
+      (userGuess == "scissors" && itemName == "stone") ||
+      (userGuess == "stone" && itemName == "paper") ||
+      (userGuess == "paper" && itemName == "scissors")
+    ) {
+      playCount = playCount + 1;
+      userWinCount = userWinCount + 1;
+      var winningRate = (userWinCount / playCount) * 100;
+      var outcomeMessage = `You win!`;
+    } // if both parties choose the same object, it's a draw.
+    else if (itemName == userGuess) {
+      playCount = playCount + 1;
+      drawCount = drawCount + 1;
+      winningRate = (userWinCount / playCount) * 100;
+      outcomeMessage = `It's a draw!`;
+    } // if the above two statements are false then the user has lost
+    else {
+      playCount = playCount + 1;
+      computerWinCount = computerWinCount + 1;
+      winningRate = (userWinCount / playCount) * 100;
+      outcomeMessage = `You lose!`;
+    }
+  }
+
+  var beginningMessage = `${userName}, the computer chose ${itemName}.<br> You chose ${userGuess}. <br> <br> ${outcomeMessage} <br> <br> Games played: ${playCount} <br> Games won: ${userWinCount} <br> Games lost: ${computerWinCount} <br> Number of draws: ${drawCount} <br> Win rate: ${winningRate}%.`;
+
+  var endingMessage = `Now you can type "scissors", "paper", "stone" to play another round!`;
+
+  if (winningRate >= 50) {
+    message = `${beginningMessage} <br> You're doing well, ${userName}! <br> <br> ${endingMessage}`;
+  } else {
+    message = `${beginningMessage} <br> You're not doing so well, ${userName}! <br> <br> ${endingMessage}`;
+  }
+
+  return message;
+};
+
+var playKoreanSPS = function (userName, userGuess) {
+  var message = "";
+  // input validation
+  if (isNaN(Number(userGuess)) == false) {
+    return message;
+  }
+  // input validation
+  if (
+    userGuess !== "scissors" &&
+    userGuess !== "paper" &&
+    userGuess !== "stone"
+  ) {
+    message = `Sorry ${userName}, there are only 3 input options available: "scissors", "paper", "stone". <br> <br> Please use lowercase letters only and try again.`;
+    return message;
+  }
```

Comment:
since the validation logic here is the same as that used in playSPS, you could try to simplify this by pulling the validation logic out as a separate function that you call inside playKoreanSPS and playSPS. This will prevent you having to repeat yourself